AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/SharedPrefHandler.java

```diff
@@ -143,6 +145,15 @@ public void validate(Element element, ElementValidation validation) {
 				}
 			}
 		}
+
+		SharedPref sharedPrefAnnotation = element.getAnnotation(SharedPref.class);
+		SharedPref.Scope scope = sharedPrefAnnotation.value();
+		String name = sharedPrefAnnotation.name();
+		boolean hasCustomName = !name.trim().isEmpty();
+		EnumSet<SharedPref.Scope> allowedScopes = EnumSet.of(SharedPref.Scope.ACTIVITY, SharedPref.Scope.UNIQUE);
+		if (hasCustomName && !allowedScopes.contains(scope)) {
+			validation.addWarning("SharedPref#name() is only supported for Scope.ACTIVITY and Scope.UNIQUE.");
```

Comment:
I would add an `error()` here. Better failing at compile time than spending hours debugging this at runtime.